### PR TITLE
[IMP] hr: give `employee` field a purpose

### PR DESCRIPTION
**Pull request by khah-odoo**
task: 4191650

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/addons/base/models/res_partner.py [hunk 1]
```diff
@@ -253,7 +253,7 @@ def default_get(self, default_fields):
     category_id: PartnerCategory = fields.Many2many('res.partner.category', column1='partner_id',
                                     column2='category_id', string='Tags', default=_default_category)
     active = fields.Boolean(default=True)
-    employee = fields.Boolean(help="Check this box if this contact is an Employee.")
+    employee = fields.Boolean(help="Whether this contact is an Employee.", compute='_compute_employee', store=True)
```
Comment (Flotchet) [inline odoo/addons/base/models/res_partner.py:256]: this should be in hr not in base IMO

**Review by Flotchet — CHANGES_REQUESTED** [review]

File: odoo/addons/base/models/res_partner.py [hunk 2]
```diff
@@ -455,6 +455,10 @@ def _compute_company_registry(self):
         for company in self:
             company.company_registry = company.company_registry
 
+    def _compute_employee(self):
+        for partner in self:
+            partner.employee = False
```
[the remark this answers is not in the document]
Comment (khah-odoo) [reply odoo/addons/base/models/res_partner.py]: Good point. Thanks!

**Review by Flotchet — APPROVED** [review]
lgtm :rocket:

File: addons/hr/models/res_partner.py [hunk 3]
```diff
@@ -9,6 +10,7 @@ class ResPartner(models.Model):
         'hr.employee', 'work_contact_id', string='Employees', groups="hr.group_hr_user",
         help="Related employees based on their private address")
     employees_count = fields.Integer(compute='_compute_employees_count', groups="hr.group_hr_user")
+    employee = fields.Boolean(help="Whether this contact is an Employee.", compute='_compute_employee', store=True)
```
Comment (tivisse) [inline addons/hr/models/res_partner.py:13]: Don't store the field, there is too many res.partner on a production db (like odoo.com), a search method is sufficient.

File: addons/hr/models/res_partner.py [hunk 4]
```diff
@@ -50,3 +52,8 @@ def _get_all_addr(self):
             'country': employee_id.private_country_id.code,
         }
         return [pstl_addr] + super()._get_all_addr()
+
+    @depends('employee_ids')
+    def _compute_employee(self):
```
Comment (tivisse) [inline addons/hr/models/res_partner.py:57]: Use read_group, and at least use the field into the partner search view, so that it gets a real purpose.

**Review by tivisse — CHANGES_REQUESTED** [review]
Small comments, thanks for your work !

File: addons/hr/models/res_partner.py (hunk 3 above)
Comment (khah-odoo) [reply addons/hr/models/res_partner.py]: Hello, the field is stored in the base module. Should I make it non-stored in hr? Also making it non-stored will cause read_group to not work properly, so I think it is better to not change it. What do you think?

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**